control_protocol.md

```diff
@@ -280,28 +280,55 @@ For the format of the Messages, see {ref}`message-layer`.
 
 ##### Coordinator sign in
 
-:::{note}
-TODO
-:::
+A Coordinator `Co1`joining a network follows a few steps:
+1. It signs in to one Coordinator `Co2` of the Network.
+2. It sends a CO_TELL_ALL message to `Co2`, to tell all other Coordinators about `Co1`s address.
+3. `Co2` tells all the Coordinators signed in (`Co3`, `Co4`...) about `Co1` with a CO_NEW message.
+4. These other Coordinators (`Co3`, `Co4`...) sign in to `Co1`.
+5. All Coordinators are connected to all others.
 
-Two Coordinators shall follow a more thorough sign in procedure, than Components:
+Two Coordinators shall follow a more thorough sign in/sign out procedure, than Components (address is for example host an port).
+The sign in might happen because of a CO_NEW message arrived or at startup.
+The sign out might happen because the Coordinator shuts down.
 
-1. Coordinator Co1 signs in to Co2
-	1. Co1 creates a DEALER socket and connects to another Coordinator Co2 (ROUTER socket).
-	2. Co1 sends a CO-SIGNIN message indicating its own address (host and port) to Co2.
-	3. Co2 creates a DEALER socket and connects to Co1's DEALER socket.
-	4. Co2 stores the the Namespace of Co1 and references it to the corresponding DEALER socket
-	5. Co2 acknowledges to Co1 the Coordinator sign in.
-2. Co2 signs in to Co1
-	6. Co2 sends a CO-SIGNIN message to Co1.
-	7. Co1 stores the Co2 Namespace and references it to the corresponding DEALER socket
-	8. Co1 ackwnowledges to Co2 the Coordinator sign in.
-3. Co1 and Co2 request the local address book of the other one.
-4. If desired (see below), Co2 tells all its known Coordinators, that they shall sign in to Co1
-
-In a CO-SIGNIN message, a Coordinator may indicate, that the other Coordinator shall tell all the other Coordinators, that it is present in the Network.
-For example Co1 asks this from Co2 and Co2 tells Co3, Co4, etc., that they shall sign in to Co1.
-That way all Coordinators are connected to all other ones.
+:::{mermaid}
+sequenceDiagram
+    participant r1 as ROUTER
+    participant d1 as DEALER
+    participant r2 as ROUTER
+    participant d2 as DEALER
+    Note over r1,d1: N1 Coordinator<br>at address1
+    Note over r2,d2: N2 Coordinator<br>at address2
+    Note over r1,d2: Sign in between two Coordinators
+    Note right of r1: shall connect<br>to address2
+    activate d1
+    Note left of d1: created with<br> name "temp-NS"
```

Comment:
why do we need this "temp-NS" in the first message? 
Why doesn't N2 respond with its name during sign-in, at which point that gets stored?

Reply:
At that initial moment, N1 does not know the Node name of N2, so you need some temporary name. N1 only knows the address of N2. 

In order to know, which Dealer socket connects to which Coordinator, we need to send some identification via round trip.
Basically we could use the conversation Id, as well, but that is not yet defined.

Reply:
Another option, which breaks the dealer to router rule:

If a Coordinator N2 receives a Co Sign in message, it responds twice: once from its router to the dealer (reverse direction), such that N1 might identify the dealer socket as connected to N2; and once from its own dealer to the other router, such that N1 can store the connection identity of N2's dealer.

Reply:
> why do we need this "temp-NS" in the first message?

Maybe that helps: we send via the Dealer and receive via the Router, so in order to know, which Dealer socket connects to wich Coordinator, we have to send some piece of information (for example that temporary dealer name) via the Dealer on a round trip, we can correlate the dealer socket to the responding (received via router) Coordinator.

Reply:
Let us discuss in #44